gsoc-2024-backend/src/main/java/backend/BackendVerticle.java

```diff
@@ -11,10 +17,46 @@ public static void main(String[] args) {
     vertx.deployVerticle(new BackendVerticle())
       .onFailure(Throwable::printStackTrace)
       .onSuccess(v -> System.out.println("Deployed backend"));
+
+  }
+  
+  public void getJoke(Promise<String > promise) {
+    WebClient client = WebClient.create(vertx);
+
+    client.getAbs("http://icanhazdadjoke.com/")
+      .putHeader("Accept", "application/json")
+      .as(BodyCodec.jsonObject())
+      .send()
+      .onSuccess(response -> {
+        promise.complete(response.body().getString("joke"));
+      })
+      .onFailure(err -> {
+        System.out.println("Something went wrong " + err.getMessage());
+        promise.fail(err);
+      });
   }
 
   @Override
   public void start(Promise<Void> startPromise) {
+
+    HttpServer server = vertx.createHttpServer();
+
+    Router router = Router.router(vertx);
+    router.route(HttpMethod.GET,"/joke").handler(ctx -> {
+      HttpServerResponse response = ctx.response();
+      response.putHeader("content-type", "text/plain");
+      Promise<String> promise = Promise.promise();
+      getJoke(promise);
+      promise.future().onComplete(result ->{
+        if(result.succeeded()){
+          response.end(result.result());
+        } else {
+          response.end("Failed to fetch joke: " + result.cause().getMessage());
+        }
+      });
+    });
+
+    server.requestHandler(router).listen(8080);
```

Comment:
What happens if the server does not start correctly? Can you update this to complete/fail the `startPromise` as needed?

Reply:
i will handle this case.